org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/sourcelookup/advanced/FileHashing.java

```diff
@@ -141,12 +143,17 @@ public HasherImpl(HasherImpl initial) {
 		}
 
 		@Override
-		public Object hash(File file) {
-			if (file == null || !file.isFile()) {
+		public HashCode hash(File file) {
+			if (file == null) {
 				return null;
 			}
 			try {
-				CacheKey cacheKey = new CacheKey(file);
+				BasicFileAttributes attributes = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+				if (!attributes.isRegularFile()) {
+					return null;
+				}
+				Object key= file.getAbsoluteFile().toPath().toAbsolutePath().normalize();
```

Comment:
Isn't better we check for empty path ?

Reply:
this is a java.nio.file.Path which does not have a a special case for empty. it can't be null either (checked in line 147)